core/src/main/java/org/opensearch/sql/expression/function/OpenSearchFunctions.java

```diff
@@ -81,10 +81,10 @@ private static Map<FunctionSignature, FunctionBuilder> getRelevanceFunctionSigna
       FunctionName funcName, int numOptionalParameters, ExprCoreType firstArgType) {
     FunctionBuilder buildFunction = args -> new OpenSearchFunction(funcName, args);
     var signatureMapBuilder = ImmutableMap.<FunctionSignature, FunctionBuilder>builder();
-    for (int numParameters = MIN_NUM_PARAMETERS - 1;
+    for (int numParameters = MIN_NUM_PARAMETERS;
          numParameters <= MIN_NUM_PARAMETERS + numOptionalParameters;
          numParameters++) {
-      List<ExprType> args = new ArrayList<>(Collections.nCopies(numParameters, STRING));
+      List<ExprType> args = new ArrayList<>(Collections.nCopies(numParameters - 1, STRING));
```

Comment:
I add one more parameter on the next line. It may be of another type, so I introduced additional function argument.
I agree that it is an ugly implementation.